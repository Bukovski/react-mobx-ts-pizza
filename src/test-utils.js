import React from 'react';
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Provider } from "mobx-react";
import { render } from "@testing-library/react";
import { rootStore } from "./store";


export const renderWithRouter = (component) => {
	const history = createMemoryHistory();
	
	return {
		...render (
			<Router history={ history }>
				{ component }
			</Router>
		)
	}
}

export const renderWithRouterMobx = (component,
                                     { initialState } = {},
                                     store = {
	...rootStore, ...initialState }) => {
	const history = createMemoryHistory()
	
	return {
		...render(
			<Router history={ history }>
				<Provider { ...store }>
					{ component }
				</Provider>
			</Router>
		),
		store
	}
}
