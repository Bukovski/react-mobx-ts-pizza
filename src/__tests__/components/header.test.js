import React from 'react';
import { cleanup, screen } from "@testing-library/react";
import { renderWithRouterMobx } from "../../test-utils";
import Header from "../../components/header.component";


describe("Header Component", () => {
	afterEach(cleanup);
	
	it("should render with Redux cart state", () => {
		const { container } =
			renderWithRouterMobx(
					<Header/>
					,{ store: {
							cart: {
								totalPrice: 1375,
								totalCount: 5
							}
						}
					}
			)
		
		expect(screen.getByText("React Pizza")).toBeInTheDocument();
		expect(screen.getByText('1375 â‚½').closest('span')).toBeInTheDocument()
		expect(screen.getByText('5').closest('span')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
});
