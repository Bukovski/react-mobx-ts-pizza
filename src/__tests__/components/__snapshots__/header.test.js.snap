// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render with Redux cart state 1`] = `
<header
  class="header"
>
  <div
    class="container"
  >
    <a
      href="/"
    >
      <div
        class="header__logo"
      >
        <img
          alt="Pizza logo"
          src="pizza-logo.svg"
          width="38"
        />
        <div>
          <h1>
            React Pizza
          </h1>
          <p>
            самая вкусная пицца во вселенной
          </p>
        </div>
      </div>
    </a>
    <div
      class="header__cart"
    >
      <a
        href="/cart"
      >
        <button
          class="button button--cart"
        >
          <span>
            1375
             ₽
          </span>
          <div
            class="button__delimiter"
          />
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.33333 16.3333C7.06971 16.3333 7.66667 15.7364 7.66667 15C7.66667 14.2636 7.06971 13.6667 6.33333 13.6667C5.59695 13.6667 5 14.2636 5 15C5 15.7364 5.59695 16.3333 6.33333 16.3333Z"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.8"
            />
            <path
              d="M14.3333 16.3333C15.0697 16.3333 15.6667 15.7364 15.6667 15C15.6667 14.2636 15.0697 13.6667 14.3333 13.6667C13.597 13.6667 13 14.2636 13 15C13 15.7364 13.597 16.3333 14.3333 16.3333Z"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.8"
            />
            <path
              d="M4.78002 4.99999H16.3334L15.2134 10.5933C15.1524 10.9003 14.9854 11.176 14.7417 11.3722C14.4979 11.5684 14.1929 11.6727 13.88 11.6667H6.83335C6.50781 11.6694 6.1925 11.553 5.94689 11.3393C5.70128 11.1256 5.54233 10.8295 5.50002 10.5067L4.48669 2.82666C4.44466 2.50615 4.28764 2.21182 4.04482 1.99844C3.80201 1.78505 3.48994 1.66715 3.16669 1.66666H1.66669"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.8"
            />
          </svg>
          <span>
            5
          </span>
        </button>
      </a>
    </div>
  </div>
</header>
`;
